components/push/src/internal/push_manager.rs

```diff
@@ -226,7 +226,10 @@ impl<Co: Connection, Cr: Cryptography, S: Storage> PushManager<Co, Cr, S> {
         if force_verify {
             self.verify_connection_rate_limiter.reset(&self.store);
         }
-        if !self.verify_connection_rate_limiter.check(&self.store) {
+
+        // If we were rate limited or there are no subscriptions yet, we should signal to the
+        // consumer that everything is ok
+        if !self.verify_connection_rate_limiter.check(&self.store) || self.uaid.is_none() {
```

Comment:
Should this check count towards the rate limiter if we don't actually make a request?  I'm mostly wondering about this and don't have a strong opinion, what do you think?

Reply:
Nope, good catch - the order of the || is off!

Reply:
and on that note, I'm openning a ticket to re-evaluate our rate-limiting strategy here, the interface has side-effects, and it's not clear the persistence to disk is exactly correct when it comes to local clocks

Reply:
https://bugzilla.mozilla.org/show_bug.cgi?id=1873237